packer/files/.vimrc

```diff
@@ -0,0 +1,151 @@
+set showmatch
+
+" Affiche le nombre de lignes sélectionnées en mode visuel ou la touche/commande qu'on vient de taper en mode commande
+set showcmd
+
+" Déplace le curseur au fur et a mesure qu'on tape une recherche, pas toujours pratique, j'ai abandonné
+"set incsearch
+
+" Utilise la souris pour les terminaux qui le peuvent (tous ?)
+" pratique si on est habitué à coller sous la souris et pas sous le curseur, attention fonctionnement inhabituel
+"set mouse=a
+
+" A utiliser en live, paste désactive l'indentation automatique (entre autre) et nopaste le contraire
+set nopaste
+
+" Indiquer le nombre de modification lorsqu'il y en a plus de 0 suite à une commande
+set report=0
+
+" Met en évidence TOUS les résultats d'une recherche, A consommer avec modération
+set hlsearch
+
+" Crée des fichiers ~ un peu partout ...
+set backup
+
+" La ruse de sioux pour ne pas qu'ils soient partout (à vous de faire le mkdir)
+" En général n'édite pas 2 fichiers de même noms fréquemment dans des répertoires différents, sinon évitez
+" -> voir by eric plus bas
+
+" Laisse les lignes déborder de l'écran si besoin
+"set nowrap
+
+" Spécial développeurs
+"
+" Indispensable pour ne pas tout casser avec ce qui va suivre
+set preserveindent
+
+" indentation automatique
+set autoindent
+set smartindent
+" Largeur de l'autoindentation
+set shiftwidth=2
+" Arrondit la valeur de l'indentation
+set shiftround
+" Largeur du caractère tab
+set tabstop=2
+" Largeur de l'indentation de la touche tab
+set softtabstop=2
+" Remplace les tab par des espaces
+set expandtab
+
+" Utilise shiftwidth à la place de tabstop en début de ligne (et backspace supprime d'un coup si ce sont des espaces)
+set smarttab
+
+" Sauvegarder les fichier ~ dans ~/.vim/backup avec crréation du répertoire si celui-ci n'existe pas
+if filewritable(expand("~/.vim/backup")) == 2
+  set backupdir=$HOME/.vim/backup
+else
+  if has("unix") || has("win32unix")
+    call system("mkdir -p $HOME/.vim/backup")
+    set backupdir=$HOME/.vim/backup
+  endif
+endif
+
+" donner des droits d'exécution si le fichier commence par #! et contient /bin/ dans son chemin
+function ModeChange()
+  if getline(1) =~ "^#!"
+    if getline(1) =~ "/bin/"
+      silent !chmod a+x <afile>
+    endif
+  endif
+endfunction
+
+au BufWritePost * call ModeChange()
+
+" Toujours laisser des lignes visibles (içi 3) au dessus/en dessous du curseur quand on
+" atteint le début ou la fin de l'écran :
+set scrolloff=3
+
+" Afficher en permanence la barre d'état (en plus de la barre de commande) :
+set laststatus=2
+
+" Format de la barre d'état (tronquée au début, fichier, flags,  :
+set statusline=%<%f%m\ %r\ %h\ %w%=%l,%c\ %p%%
+
+" indentation cfengine
+filetype plugin on
+":helptags ~/.vim/cf3_doc/
+au BufRead,BufNewFile *.cf set ft=cf3
+au BufRead,BufNewFile *.cf.2 set ft=cf3
+au BufRead,BufNewFile *.st set ft=cf3
+au BufRead,BufNewFile *.st.2 set ft=cf3
```

Comment:
Could you add the following:
  au BufRead,BufNewFile *.ps1 set ft=ps1
  au BufRead,BufNewFile *.ps1.st set ft=ps1